### PR TITLE
[x64] make jnp.histogram and related functions work with strict promotion

### PR DESCRIPTION
**Pull request by jakevdp**
Re-lands #10907 after rollback in #10909

### CODE REVIEW

**Comment by yashk2810** [comment]
My rollback just went in. I have applied the kokoro label. If all tests pass, I'll approve or you can apply the pull ready label directly (if I am offline)

**Review by yashk2810 — APPROVED** [review]
